core/trino-main/src/main/java/io/trino/sql/planner/SortExpressionExtractor.java

```diff
@@ -124,13 +124,20 @@ protected Optional<SortExpressionContext> visitComparison(Comparison comparison,
         }
 
         @Override
-        protected Optional<SortExpressionContext> visitBetween(Between node, Void context)
+        protected Optional<List<SortExpressionContext>> visitBetween(Between node, Void context)
         {
-            Optional<SortExpressionContext> result = visitComparison(new Comparison(GREATER_THAN_OR_EQUAL, node.value(), node.min()), context);
-            if (result.isPresent()) {
-                return result;
+            // Handle both side of BETWEEN as `GREATER_THAN_OR_EQUAL` expression and `LESS_THAN_OR_EQUAL` expression.
```

Comment:
this will be super easy if return value is list:
```
return ImmutableList.builder()
 .addAll(visitComparison(new Comparison(Comparison.Operator.GREATER_THAN_OR_EQUAL, node.value(), node.min()), context))
 .addAll(visitComparison(new Comparison(Comparison.Operator.LESS_THAN_OR_EQUAL, node.value(), node.max()), context)
 .build();
```